src/Backends/PrefersColorSchemeSettings.vala

```diff
@@ -0,0 +1,144 @@
+/*
+* Copyright 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+*/
+
+public class SettingsDaemon.Backends.PrefersColorSchemeSettings : GLib.Object {
+    public unowned PantheonShell.Pantheon.AccountsService accounts_service { get; construct; }
+
+    private GLib.Settings color_settings;
+    private double pos_lat = -1.0;
+    private double pos_long = -1.0;
+
+    private AsyncResult async_result;
+    private bool time_running = false;
+    private uint time_id;
+
+    public PrefersColorSchemeSettings (PantheonShell.Pantheon.AccountsService accounts_service) {
+        Object (accounts_service: accounts_service);
+    }
+
+    construct {
+        color_settings = new GLib.Settings ("io.elementary.settings-daemon.prefers-color-scheme");
+
+        color_settings.changed["prefer-dark-schedule"].connect (update);
+
+        update ();
+    }
+
+    private void update () {
+        var schedule = color_settings.get_string ("prefer-dark-schedule");
+
+        if (schedule == "sunset-to-sunrise") {
+            get_location.begin ((obj, res) => {
+                async_result = res;
+            });
+
+            start_timer ();
+        } else if (schedule == "manual") {
+            start_timer ();
+        } else {
+            get_location.end (async_result);
+            stop_timer ();
+        }
+    }
+
+    private void start_timer () {
+        if (!time_running) {
```

Comment:
Instead of using a separate `time_running` boolean, you could just set `time_id` to 0 initially and every time the timer is stopped/removed and then check that to see if the timer is attached or not.